cartographer_ros/cartographer_ros/node.cc

```diff
@@ -261,6 +261,16 @@ void Node::PublishLocalTrajectoryData(const ::ros::TimerEvent& timer_event) {
         node_options_.use_pose_extrapolator
             ? ToRos(now)
             : ToRos(trajectory_data.local_slam_data->time);
```

Comment:
I think it would be better to change this here and just use
```c++
stamped_transform.header.stamp = ToRos(now)
```
always.

I experienced similar timing issues when using Cartographer with `use_pose_extrapolator == false` in a Gazebo simulator and solved it like this (patch is currently only in a private fork of this repo).

Reply:
I just tried this fix and it doesn't work for me, warning is happening again

Reply:
Hi Michael, I think that the data should be accurately stamped - if you're using the external pose extrapolator, the published extrapolated pose data should have a newer stamp; otherwise, it should be stamped as it is, the last lidar pose coming from local slam. And with a high enough publishing frequency, you can run into the same stamp if you have a `use_sim_time` clock.

The solution is that wherever you're using the external pose extrapolator or not, the stamp of the published poses should be enforced to be scrictly increasing. I think this PR accurately represents this intent.

@stribor14 you're missing a dot at the second row of the comment.

Reply:
@ojura Thanks for the clarification. I would be fine with merging this, I will only add a few comments for smaller nits.